invenio_app_ils/documents/api.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""ILS Document APIs."""
+
+from functools import partial
+
+from flask import current_app
+from invenio_circulation.search.api import search_by_pid
+from invenio_pidstore.models import PIDStatus
+from invenio_pidstore.providers.recordid_v2 import RecordIdProviderV2
+
+from invenio_app_ils.errors import RecordHasReferencesError
+from invenio_app_ils.fetchers import pid_fetcher
+from invenio_app_ils.minters import pid_minter
+from invenio_app_ils.proxies import current_app_ils
+from invenio_app_ils.records.api import IlsRecordWithRelations
+
+DOCUMENT_PID_TYPE = "docid"
+DOCUMENT_PID_MINTER = "docid"
+DOCUMENT_PID_FETCHER = "docid"
+
+DocumentIdProvider = type(
```

Comment:
For reviewers, this is the only real change: inline minters/fetcher instead of having multiple files (minter, fetcher, provider, pid). Do you find it better?

Reply:
Personally I like this approach so I don't have to jump to different files to see these values :) My only concern is if in the future there is the need of adding some custom code on these providers then the `type` approach would not be in handy compared to a class definition. But this can easily be refactored whenever and if is needed :)

Reply:
It is more organised with less noise for sure 👍